lib/decoder_impl.cc

```diff
@@ -331,23 +335,20 @@ namespace gr {
                 return 0.0f;
             }
 
-            max = std::max_element(samples + gr::lora::clamp((int)(max - coeff),  0, (int)window),
-                                   samples + gr::lora::clamp(max +        coeff, 0u,      window)) - samples;
-            min = std::min_element(samples + gr::lora::clamp(max +           1u, 0u,      window),
-                                   samples + gr::lora::clamp(max +   2u * coeff, 0u,      window)) - samples;
-
-            max += (min - max) / 2;
-            *index = max;
-
-            // Extra to allow falling edge on 99% of sample == Practically synced
-            // Should improve detection by not discarding almost synced samples, but doesn't..
-//            if ((max - coeff / 2) > (window * 0.98f)) {
-////                printf("ALREADY SYNCED\n");
-//                *index = 0;
-//                return 0.99f;
-//            }
+            max = std::max_element(samples + gr::lora::clamp((int)(max -  2u * coeff),  0, (int)window),
```

Comment:
Naming of ```max``` and ```min``` should be changed to be less confusing: the variable ```max``` is for example reused for different purposes (as some value in the middle, as the maximum value of the symbol, and as the argmax of this value), making the code hard to read.